packages/components/src/components/audio/audio.ts

```diff
@@ -0,0 +1,539 @@
+        damping: 1
+      }),
+      new Wave({
+        canvas: this.canvas,
+        color: computedColor,
+        phase: 180,
+        shift: -2.5,
+        amplitude: 250,
+        frequency: 0.005,
+        damping: 1
+      }),
+      new Wave({
+        canvas: this.canvas,
+        color: computedColor,
+        phase: 180,
+        shift: -1.5,
+        amplitude: 200,
+        frequency: 0.01,
+        damping: 1
+      }),
+      new Wave({
+        canvas: this.canvas,
+        color: computedColor,
+        phase: 180,
+        shift: 1.5,
+        amplitude: 150,
+        frequency: 0.01,
+        damping: 1
+      })
+    ];
+    this.draw();
+  }
+
+  private stopAnimation() {
+    this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+    this.waveList = [];
+  }
+
+  private clear() {
+    this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+  }
+
+  async draw() {
+    if (!this.animated && !this.isPlaying) return;
+
+    this.clear();
+    this.waveList.forEach(wave => {
+      wave.redraw();
+    });
+    await new Promise(resolve => {
+      setTimeout(resolve, 1000 / 30);
+    });
+    await this.draw();
+  }
+
+  render() {
+    const progressPercentage = this.audioElement ? (this.progress / this.audioElement.duration) * 100 : 0;
+
+    const renderAudioControls = html`<div
+      class=${cx(
+        'audio-player__controls grid grid-cols-3 justify-items-center items-center',
+        !this.animated && 'relative',
+        this.animated && !this.reversedLayout && 'absolute top-0 left-0 w-full',
+        this.reversedLayout ? 'mt-2' : 'mb-2'
+      )}
+      part="audio-controls"
+    >
+      <button
+        class=${cx(
+          'audio-player__playback-speed justify-self-start text-base font-bold hover:cursor-pointer hover:text-primary-500',
+          this.inverted ? 'text-white focus-visible:focus-outline-inverted' : 'text-primary focus-visible:focus-outline'
+        )}
+        aria-label="${this.localize.term('playbackSpeed')}"
+        tabindex="0"
+        @click=${this.togglePlaybackSpeed}
+        @keydown=${this.togglePlaybackSpeedKeydown}
+        part="playback-speed"
+      >
+        ${this.speed}x
+      </button>
+
+      <button
+        class=${cx(
+          'flex justify-center items-center p-4 rounded-full cursor-pointer hover:cursor-pointer hover:bg-primary-500',
+          this.inverted ? 'bg-white focus-visible:focus-outline-inverted' : 'bg-primary focus-visible:focus-outline'
+        )}
+        part="play-button"
+        @click=${!this.isPlaying ? this.playAudio : this.pauseAudio}
+        aria-label="${this.isPlaying ? this.localize.term('pauseAudio') : this.localize.term('playAudio')}"
+        tabindex="0"
+      >
+        ${this.isPlaying
+          ? html`<slot name="pause-icon">
+              <sd-icon
+                name="pause"
+                library="system"
+                class=${cx('w-6 h-6', this.inverted ? 'text-primary' : 'text-white')}
```

Comment:
You could set the color just on the button above, so you don't have to redefine it twice. This furthermore would make it easier if people slot other elements as well. Same for all other elements.

Reply:
Updated